src/api/connectors/AppleConnector.ts

```diff
@@ -90,35 +91,48 @@ class AppleConnector implements ConnectorHandler {
             const payloadData = payload.data as AppleShowTrendsFollowersPayload
 
             // map all day values into one structure
-            let days = payloadData.followerAllTimeTrends.reduce(
+            let followerData = payloadData.followerAllTimeTrends.reduce(
                 (prev, curr) => {
-                    prev[curr[0]] = {
-                        date: curr[0],
-                        totalListeners: curr[1],
+                    // use moment to get date from YYYYMMDD format
+                    const day = curr[0].toString()
+                    const totalFollowers = curr[1]
+                    const totalUnfollowers = curr[2]
+                    prev[day] = {
+                        date: day,
+                        totalFollowers: totalFollowers,
+                        totalUnfollowers: totalUnfollowers,
                         gained: 0,
                         lost: 0,
                     } as AppleShowTrendsFollowersDay
                     return prev
                 },
                 {} as {
-                    [day: number]: AppleShowTrendsFollowersDay
+                    [day: string]: AppleShowTrendsFollowersDay
                 }
             )
 
             // add gained/lost values
-            days = payloadData.followerGrowthTrends.reduce((prev, curr) => {
-                const dayElem = prev[curr[0]] || null
-                if (dayElem) {
-                    dayElem.gained = curr[1]
-                    dayElem.lost = curr[1]
-                }
-                return prev
-            }, days)
+            followerData = payloadData.followerGrowthTrends.reduce(
+                (prev, curr) => {
+                    // as gained/lost is the difference between the date and the next date
+                    // we want to store it together with the next day
```

Comment:
That comment confuses me. Shouldn't it be
```suggestion
                    // we want to store it together with the previous day
```
? 🤔 Depends on how you look at it, I guess.

Reply:
yup, but gained has to be stored with the next day